testsuite/MDAnalysisTests/coordinates/test_h5md.py

```diff
@@ -493,9 +358,325 @@ def test_no_units(h5md_file, ref, tmpdir):
+                                timeunit, lengthunit,
+                                velocityunit, forceunit):
+        with Writer(outfile,
+                    universe.atoms.n_atoms,
+                    velocities=True,
+                    forces=True,
+                    lengthunit=lengthunit,
+                    velocityunit=velocityunit,
+                    forceunit=forceunit,
+                    timeunit=timeunit) as W:
+            for ts in universe.trajectory:
+                W.write(universe)
+
+        u = mda.Universe(TPR_xvf, outfile)
+        for u_unit, custom_unit in zip (u.trajectory.units.values(),
+                                       (timeunit, lengthunit,
+                                       velocityunit, forceunit)):
+            assert_equal(u_unit, custom_unit)
+
+    @pytest.mark.parametrize('timeunit, lengthunit, velocityunit, forceunit', (
+        ('imaginary time', None, None, None),
+        (None, None, 'c', None),
+        (None, None, None, 'HUGE FORCE',),
+        (None, 'lightyear', None, None)))
+    def test_write_bad_units(self, universe, outfile, Writer,
+                             timeunit, lengthunit,
+                             velocityunit, forceunit):
+        with pytest.raises(RuntimeError):
+            with Writer(outfile,
+                        universe.atoms.n_atoms,
+                        velocities=True,
+                        forces=True,
+                        lengthunit=lengthunit,
+                        velocityunit=velocityunit,
+                        forceunit=forceunit,
+                        timeunit=timeunit) as W:
+                for ts in universe.trajectory:
+                    W.write(universe)
+
+    @pytest.mark.parametrize('convert_units, units', (
+        (True, ['ps', 'nm', 'nm/ps', 'kJ/(mol*nm)']),
+        (False, ['ps', 'Angstrom', 'Angstrom/ps', 'kJ/(mol*Angstrom)'])))
+    def test_convert_units(self, universe, outfile, Writer,
+                           convert_units, units):
+        with Writer(outfile,
+                    universe.atoms.n_atoms,
+                    velocities=True,
+                    forces=True,
+                    convert_units=convert_units) as W:
+            for ts in universe.trajectory:
+                W.write(universe)
+
+        uw = mda.Universe(TPR_xvf, outfile)
+        uw_units = uw.trajectory.units.values()
+        for u1, u2 in zip(units, uw_units):
+            assert_equal(u1, u2)
+
+    #@pytest.mark.parametrize('chunks', (None, (1, 1000, 3)))
+    def test_write_chunks(self, universe, outfile, Writer):
+        chunks = (1, 1000, 3)
+        with Writer(outfile,
+                    universe.atoms.n_atoms,
+                    chunks=chunks) as W:
+            for ts in universe.trajectory:
+                W.write(universe)
+
+        uw = mda.Universe(TPR_xvf, outfile)
+        dset = uw.trajectory._particle_group['position/value']
+        assert_equal(dset.chunks, chunks)
```

Comment:
Should you also assert actual values to make sure that they end up in the correct position in the file?